Desktop/AddGoalWindow.cs

```diff
@@ -51,78 +53,138 @@ private void PopulateGoalListView()
             }
         }
 
-        public void ValidateFields(string amount, string details)
+        public void ValidateFields(string amount, string details, string name)
         {
             if (String.IsNullOrWhiteSpace(amount))
             {
-                goalMoney.BackColor = Color.Red;
+                  goalMoney.BackColor = Color.Red;
             }
+
             if (String.IsNullOrWhiteSpace(details))
             {
-                goalMoney.BackColor = Color.Red;
+                descriptionBox.BackColor = Color.Red;
+            }
+
+            if (String.IsNullOrWhiteSpace(name))
+            {
+                goalNameBox.BackColor = Color.Red;
             }
         }
+
         private void button1_Click(object sender, EventArgs e)
         {
-            DateTime date = goalDate.Value;
-            string amount = goalMoney.Text;
-            string name = goalNameBox.Text;
-            string description = descriptionBox.Text;
-
-            ValidateFields(amount, name);
-
-            if (!String.IsNullOrWhiteSpace(amount) && !String.IsNullOrWhiteSpace(name))
+            var date = goalDate.Value;
+            var amount = goalMoney.Text;
+            var name = goalNameBox.Text;
+            var description = descriptionBox.Text;
+            decimal amountInDecimal;
+            ValidateFields(amount, description, name);
+
+            if (!String.IsNullOrWhiteSpace(amount) && !String.IsNullOrWhiteSpace(name) && decimal.TryParse(amount, out amountInDecimal) && date.ToShortDateString() != DateTime.UtcNow.ToShortDateString())
             {
-                decimal amountInDecimal = decimal.Parse(amount);
-
-                Database db = new Database();
+                MessageBox.Show(date.ToShortDateString());
+                var db = new Database();
 
-                Goal newGoal = new Goal
+                var newGoal = new Goal
                 {
-                    GoalDate = date,
+                    Deadlinedate = date,
                     Amount = amountInDecimal,
                     Title = name,
-                    Description = description
+                    Description = description,
+                    Creationdate = DateTime.UtcNow
                 };
                 db.AddGoal(newGoal);
                 UpdateGoalList();
             }
-
+            else
+            {
+                MessageBox.Show("Wrong format");
+            }
         }
+
         private void backButton_Click(object sender, EventArgs e)
         {
-            this.Close();
+            Close();
+        }
+
+        private void DeleteButton_Click(object sender, EventArgs e)
+        {
+            var selectedTitle = listView1.Items[selectedId].Text;
+            listView1.Items.RemoveAt(selectedId);
+            db.RemoveGoal(selectedTitle);
+            UpdateGoalList();
+        }
+
+        private void listView1_SelectedIndexChanged(object sender, EventArgs e)
+        {
+
         }
```

Comment:
What is the purpose of this method?

Reply:
This is automatically made by Visual studio, it has no purpose, i will delete it.